4-data-structures.md

```diff
@@ -0,0 +1,97 @@
+# Data Structures in Swift
+
+This post will be focusing on tuples, arrays, and dictionaries, although other data 
+structures such as sets also exist. Of these, arrays have been present in programming
+for a long time - at least since the development of C, and probably before that (I'm a 
+little rusty on my history). Dictionaries were part of Objective-C, which is the language
+I'm most familiar with, and tuples are becoming more and more common in newer languages 
+such as python and Ruby. In particular, one of the large differences I'm seeing in Swift
+from Objective-C is that *it's not C-derived any more*. Objective-C had very clear
+influences from its mother language, and actually maintained almost all primitive C
+functionality, from arrays to functions, in addition to what it built on top of that. You
+could still use simple bracketed arrays, even though the language encouraged use of the
+```NSArray``` objects, as well as old-style functions in the form of 
+```int function(int var1, int var2)``` rather than Objective-C's 
+```- (int) function: (int) var1 withVariable: (int) var2```. Other than syntax, 
+however, Swift's method declarations are similar in structure to those of Objective-C.  
```

Comment:
When you give code examples, please put them on their own paragraphs, so they are code blocks rather than mid-sentence code snippets. You should use shorter paragraphs and more code.

An example of a code block on its own:

```swift
func foo {
  println("Hello, world")
}
```